src/actions/collection.ts

```diff
@@ -0,0 +1,75 @@
+import { Action } from '@ngrx/store';
+import { Book } from '../models/book';
+import { label } from '../util';
+
+
+export const CollectionActionTypes = {
+  ADD_BOOK: label('[Collection] Add Book'),
+  ADD_BOOK_SUCCESS: label('[Collection] Add Book Success'),
+  ADD_TO_COLLECTION_FAIL: label('[Collection] Add Book Fail'),
+  REMOVE_BOOK: label('[Collection] Remove Book'),
+  REMOVE_BOOK_SUCCESS: label('[Collection] Remove Book Success'),
+  REMOVE_BOOK_FAIL: label('[Collection] Remove Book Fail'),
+  LOAD_COLLECTION: label('[Collection] Load Collection'),
+  LOAD_COLLECTION_SUCCESS: label('[Collection] Load Collection Success')
+};
+
+
+export class AddBook implements Action {
```

Comment:
This doesn't seem very DRY... What would be an example of any methods that would go in these classes?  Or is the use of classes here just for typesafety when using the actions in a reducer?